packager/mpd/base/representation.cc

```diff
@@ -271,8 +268,11 @@ base::Optional<xml::XmlNode> Representation::GetXml() {
   }
 
   if (HasLiveOnlyFields(media_info_) &&
-      !representation.AddLiveOnlyInfo(media_info_, segment_infos_,
-                                      start_number_)) {
+      !representation.AddLiveOnlyInfo(
```

Comment:
I'd update `AddLiveOnlyInfo` to remove the `start_number_` parameter instead.

Reply:
And getting the start_number in the method? 
```
bool RepresentationXmlNode::AddLiveOnlyInfo(
    const MediaInfo& media_info,
    const std::list<SegmentInfo>& segment_infos) {
  XmlNode segment_template("SegmentTemplate");

  int start_number = segment_infos.empty()? 
	  1 : segment_infos.begin()->start_segment_index;
```

Reply:
Or perhaps adding start_number only if segment_infos is not empty, e.g.

```
  if (media_info.has_segment_template_url() && !segment_infos.empty()) {
```